### PR TITLE
Pre release hello_helpers

### PR DESCRIPTION
**Pull request by hello-chintan**
This PR ports all pending features of the hello_helpers package

Notable changes include:
- Class HelloNode ported to ROS 2:
    - The node uses a MultiThreadedExecutor()
    - All topics, services and callbacks use a ReentrantCallback() group
    - Added utility functions to call the following services:
        - home_the_robot
        - stow_the_robot
        - switch_to_position_mode
        - switch_to_trajectory_mode
        - switch_to_navigation_mode
- Documentation README added

To test, follow the examples in the README.

### CODE REVIEW

File: hello_helpers/README.md [hunk 1]
```diff
@@ -2,24 +2,281 @@
 
 ## Overview
 
-*hello_helpers* mostly consists of the hello_helpers Python module. This module provides various Python files used across stretch_ros2 that have not attained sufficient status to stand on their own.
+*hello_helpers* mostly consists of the hello_helpers Python module. This module provides various Python files used across stretch_ros that have not attained sufficient status to stand on their own.
```
Comment (hello-binit) [inline hello_helpers/README.md]: stretch_ros becomes stretch_ros2

File: hello_helpers/src/hello_helpers/hello_misc.py [hunk 2]
```diff
@@ -104,75 +108,249 @@ def get_robot_floor_pose_xya(tf2_buffer, floor_frame='odom'):
...
+            self.node.get_logger().error('Unable to connect to arm action server. Timeout exceeded.')
             sys.exit()
         
         self.tf2_buffer = tf2_ros.Buffer()
-        self.tf2_listener = tf2_ros.TransformListener(self.tf2_buffer, self)
+        self.tf2_listener = tf2_ros.TransformListener(self.tf2_buffer, self.node)
         
-        self.point_cloud_subscriber = self.create_subscription(PointCloud2, '/camera/depth/color/points', self.point_cloud_callback, 10)
-        self.point_cloud_pub = self.create_publisher(PointCloud2, '/' + node_topic_namespace + '/point_cloud2', 10)
+        self.tool_subscriber = self.node.create_subscription(String, '/tool', self.tool_callback, 10, callback_group=reentrant_cb)
```
Comment (hello-binit) [inline hello_helpers/src/hello_helpers/hello_misc.py:308]: Do we have a /tool topic in Stretch Driver?

**Comment by hello-binit** [comment]
Will we switch over the demo nodes to use `HelloNode` in this PR or a separate PR?

**Comment by hello-chintan** [comment]
@hello-atharva was able to circumvent the HelloNode dependency for the demos. Do we want to preserve the dependency and revert the demos back to using HelloNode? If so, a separate PR would be ideal.

File: hello_helpers/src/hello_helpers/hello_misc.py (hunk 2 above)
Comment (hello-chintan) [reply hello_helpers/src/hello_helpers/hello_misc.py:308]: Yes, we do!

File: hello_helpers/README.md (hunk 1 above)
Comment (hello-chintan) [reply hello_helpers/README.md]: Ah! Good catch.

**Comment by hello-atharva** [comment]
@hello-chintan I think it would be good to preserve the dependency so as to show users how to use the base class.

**Review by hello-atharva — APPROVED** [review]

**Comment by hello-chintan** [comment]
@hello-atharva gotcha! Pointing this PR to Iron and merging.

**Pull request merged**